scripts/admin/check-legacy-syntax/check-syntax.js

```diff
@@ -0,0 +1,215 @@
+    serverFiles: ["versions.js"],
+  },
+  "ecmascript-runtime-server": {},
+  "es5-shim": {},
+  fetch: {},
+  "geojson-utils": {},
+  "hot-code-push": {},
+  "hot-module-replacement": {},
+  insecure: {},
+  "inter-process-messaging": {
+    serverFiles: ["inter-process-messaging.js"],
+  },
+  "launch-screen": {},
+  localstorage: {},
+  "logic-solver": {
+    // TODO: Fibers - Legacy
+    // Revisit when we remove fibers, this may break for legacy
+    ignoredFiles: ["logic.js", "optimize.js"],
+  },
+  "meteor-base": {},
+  "mobile-experience": {},
+  "mobile-status-bar": {},
+  "modern-browsers": {
+    serverFiles: ["modern.js"],
+  },
+  modules: {},
+  "modules-runtime": {},
+  "modules-runtime-hot": {},
+  "mongo-dev-server": {},
+  "mongo-livedata": {},
+  "npm-mongo": {
+    serverFiles: ["wrapper.js"],
+  },
+  "package-stats-opt-out": {},
+  "package-version-parser": {},
+  promise: {},
+  "react-fast-refresh": {},
+  "reactive-var": {},
+  "reload-safetybelt": {
+    // is a server only file that uses TLA.
+    ignoredFiles: ["reload-safety-belt.js"],
+  },
+  sha: {},
+  "standard-minifiers": {},
+  // 'test-in-console': {},
+  "test-server-tests-in-console-once": {},
+  "tinytest-harness": {},
+  "twitter-config-ui": {},
+  // 'twitter-oauth': {},
+  typescript: {
+    serverFiles: ["plugin.js"],
+  },
+  underscore: {},
+  url: {},
+};
+
+const acorn = require("acorn");
+const fs = require("fs");
+const path = require("path");
+
+const baseDir = path.resolve(__dirname, "../../../");
+
+Object.keys(packages).forEach((packageName) => {
+  console.log(`=> Checking ${packageName}`);
+
+  const packagePath = path.resolve(baseDir, "packages", packageName);
+  let files = listPackageFiles(packagePath);
+
+  for (const file of files) {
+    let content = fs.readFileSync(file, "utf-8");
+    let relPath = path.relative(packagePath, file);
+    let ecmaVersion = clientEcmascriptVersion;
+
+    if (
+      packages[packageName].serverFiles?.includes(relPath) ||
+      relPath.endsWith("_server.js") ||
+      file.endsWith("/server.js")
+    ) {
+      // Is a server file, which can use some newer syntax
+      ecmaVersion = serverEcmascriptVersion;
+    }
+
+    if (packages[packageName].ignoredFiles?.includes(relPath)) {
+      // is a server file that may use a lot of newer syntax, such as TLA.
+      ecmaVersion = latestEcmascriptVersion;
+    }
+
+    try {
+      acorn.parse(content, {
+        ecmaVersion,
+      });
+    } catch (error) {
+      console.log("");
+      console.error(`Failed to parse ${file}: `, error.message);
+      let line = content.split("\n")[error.loc.line - 1];
+      console.log(line);
+      console.log("");
+
+      if (
+        error.message.includes("Unexpected token") &&
+        line.includes("await")
+      ) {
+        console.log(
+          "Failed to parse, but it's probably because you're using TLA."
+        );
+        console.log("Will continue");
+        continue;
+      }
```

Comment:
If a package is able to use TLA (has a dependency on ecmascript), it can be removed from the list of packages this script runs on. This script is only needed for packages that can't use TLA/ecmascript.